### PR TITLE
Usability changes

### PR DESCRIPTION
**Pull request by imallona**
## Ticket

This was meant to contain usability changes but ended up being the development collector.

Main changes:
- release candidate
- storage backends
- click
- software "prepare" verbs
- one bucket storage
- easybuild module updates

## Description

Many changes, as it incorporates:
- https://github.com/omnibenchmark/omnibenchmark/pull/18
- https://github.com/omnibenchmark/omnibenchmark/pull/17
- https://github.com/omnibenchmark/omnibenchmark/pull/16
- https://github.com/omnibenchmark/omnibenchmark/pull/15

## Checklist:
- [x] My code follows the code style of this project.
- [x] My CLI method respects the signature defined in the task.
- [x] I have documented the CLI method accordingly.
- [x] I have added tests to cover my changes.
- [x] I have added a CHANGELOG entry.

### CODE REVIEW

File: omni/cli/benchmark.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def diff_benchmark(
     )
 
 
-@cli.command("list versions")
+@cli.command("list-versions")
```
Comment (imallona) [inline omni/cli/benchmark.py:129]: is there documentation on how to use this? @retogerber

**Comment by imallona** [comment]
What do you think about adding a `print workflow graph`-like command to the run benchmark CLI? @DanInci You have it implemented (omni_workflow)

**Comment by imallona** [comment]
What about adding a 'license' field and 'path to the code of conduct' to the benchmarking yaml? @imallona

**Comment by imallona** [comment]
What about removing `requirements.txt` from https://github.com/omnibenchmark-example/data and other example repos ; and think of updating the config file so it resembles what agreed (contributor yaml)?

**Comment by imallona** [comment]
How are the software backends specified exactly? e.g. where in the codebase do we pass to Snakemake the `--use-conda`, `--use-singularity` etc according to the yaml?

**Comment by imallona** [comment]
I think the run-module is now broken for commit https://github.com/omnibenchmark/omnibenchmark/pull/13/commits/4ad5e93137048c285997f9492fa21f2de907e8b6 but also for the current's main head (separate comment)

```
$ ob run module --benchmark tests/data/Benchmark_001.yaml --module D1 
Error: At least one option must be specified. Use '--input', '--example', or '--all'.

$ ob run module --benchmark tests/data/Benchmark_001.yaml --module D1 --all
Running module on all available remote datasets.
Error: Remote execution is not supported yet. Workflows can only be run in local mode.

$ ob run module --benchmark tests/data/Benchmark_001.yaml --module D1 --all --local
Usage: ob run module [OPTIONS]
Try 'ob run module --help' for help.
╭─ Error ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ No such option: --local Did you mean --all?                                                                                                                │
╰───────────────────────────────────────────────────────────────────────────
```

**Comment by retogerber** [comment]
Feat: Environment variable for benchmark yaml file.
Instead of 
`ob run benchmark --benchmark tests/data/Benchmark_001.yaml`
it would be nice to just run
`ob run benchmark`

File: omni/io/files.py [hunk 2]
```diff
@@ -15,37 +18,46 @@ def list_files(
...
-            "storage_type": "minio",
-        }
-        benchmark = "testversioning"
-        version = "0.1"
-
-    ss = get_storage(bench_yaml["storage_type"], bench_yaml["auth_options"], benchmark)
-    # set version
-    ss.set_version(version)
-    # list objects of version
+    benchmark = "tests/data/Benchmark_003.yaml"
```
Comment (retogerber) [inline omni/io/files.py:24]: TODO: remove line

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        os: [ubuntu-latest] # [ubuntu-22.04, macos-latest]
         python-version: ["3.12"]
```
Comment (DanInci) [inline .github/workflows/tests.yml:16]: Should we remove the tests on macos and windows permanently?
Or have a note somewhere to make them work later on? I think at least the macos ones are necessary

File: omni/workflow/snakemake/rules/rule_node.smk [hunk 4]
```diff
@@ -28,6 +36,12 @@ def _create_initial_node(node):
             dataset=module_id
         output:
             formatter.format_output_templates_to_be_expanded(node),
+        conda:
+            _get_environment_path(benchmark, node) if backend == SoftwareBackendEnum.conda else None
+        envmodules:
+            _get_environment_path(benchmark,node) if backend == SoftwareBackendEnum.envmodules else None
+        container:
+            _get_environment_path(benchmark,node) if backend == (SoftwareBackendEnum.apptainer or SoftwareBackendEnum.docker) else None
```
Comment (DanInci) [inline omni/workflow/snakemake/rules/rule_node.smk:44]: TODO: Should we keep both "aptainer" and "docker" enum values meaning the same software backend?

**Review by DanInci — APPROVED** [review]

File: omni/io/files.py (hunk 2 above)
Comment (imallona) [reply omni/io/files.py:24]: done

File: omni/workflow/snakemake/rules/rule_node.smk (hunk 4 above)
Comment (imallona) [reply omni/workflow/snakemake/rules/rule_node.smk:44]: hmmm, snakemake does not run using docker (uses docker images, but with apptainer). We might provide a workaround though, so macos people are happier

File: .github/workflows/tests.yml (hunk 3 above)
Comment (imallona) [reply .github/workflows/tests.yml]: brought them back, thanks

Comment (imallona) [reply .github/workflows/tests.yml]: I don't think the test environment has everything needed (conda) to run these tests [trace](https://github.com/omnibenchmark/omnibenchmark/actions/runs/11007868739/job/30564473829). Perhaps we could run these using envs managed by micromamba, [as we do for software tests](https://github.com/omnibenchmark/omnibenchmark/actions/runs/11007868732/job/30564474019)?

**Pull request merged**